### PR TITLE
First pass at supercluster enablement.

### PR DESCRIPTION
**Pull request by derekcollison**
This allows metacontrollers to span superclusters. Also includes placement directives for streams. By default they select the request origin cluster.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

**Review by kozlovic — APPROVED** [review]
LGTM

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -46,6 +46,12 @@ type jetStreamCluster struct {
 	consumerResults *subscription
 }
 
+// Used to guide placement of streams in clustered JetStream.
+type Placement struct {
+	Cluster string   `json:"cluster"`
+	Tags    []string `json:"tags,omitempty"`
```
Comment (ripienaar) [inline server/jetstream_cluster.go:52]: cluster not having omitempty, does this imply that cluster is required? seems empty is valid so just checking

Comment (derekcollison) [reply server/jetstream_cluster.go:52]: We could make optional, for now it is required.

Comment (ripienaar) [reply server/jetstream_cluster.go:52]: hmm, appears to me we do handle it being empty, which looks like it then does random assignment? That would be what I'd expect anyway

Comment (derekcollison) [reply server/jetstream_cluster.go:52]: Yes if its empty we pick point of origin. What I meant was if you are filling this in you should have cluster for now.. But I can make optional or you can submit quick PR for omitempty addition.

Comment (ripienaar) [reply server/jetstream_cluster.go:52]: can wait till we do tag matching and all :)

**Pull request merged**